app/dao/mongo.go

```diff
@@ -0,0 +1,84 @@
+package dao
+
+import (
+	"fmt"
+
+	"github.com/companieshouse/payment-reconciler/app/config"
+	"github.com/companieshouse/payment-reconciler/app/models"
+	"github.com/globalsign/mgo"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// Mongo provides a MongoDB implementation of the DAO
+type Mongo struct {
+	Config *config.Config
+}
+
+// New returns a new Mongo struct using the provided config
+func New(cfg *config.Config) *Mongo {
+
+	return &Mongo{
+		Config: cfg,
+	}
+}
+
+// getMongoSession retrieves a fresh MongoDB session
+func getMongoSession(cfg *config.Config) (*mgo.Session, error) {
+
+	s, err := mgo.Dial(cfg.MongoDBURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.Copy(), nil
+}
+
+// GetTransactionsData fetches transactions data
+func (m *Mongo) GetTransactionsData() (models.TransactionsData, error) {
+
+	var transactions []models.Transaction
+
+	var transactionsData models.TransactionsData
```

Comment:
Could we name this as TransactionList instead of TransactionsData as it may get a little confusing because we have(in transaction api) mongoDB structure where we have data inside transactions.

Reply:
Renamed 👍